### DISCUSSION TITLE
Dynamically invoke a machine

### DISCUSSION THREAD

**Discussion by martinschneider01** [statelyai/xstate#4415] [Q&A] [answered]
Hi!

I'm a really beginning user of xState, I was wondering if something like that would be possible in v5:

I want to invoke a machine based on a parameter sent on an event. In a near future, I would like to choose between 8-10 different machines. It would be lovely if I could use invoke API such as "onDone" (when child machine reach an end state) or "onError"

I saw that my question was [already asked a few years ago](https://github.com/statelyai/xstate/issues/2692), but I wonder if since v5 there is a better way to do that? Tried to look into v5 doc but I haven't found (or didn't understood fully) any solution

**Comment by zavx0z** [comment, marked as answer]
[issues/4410](https://github.com/statelyai/xstate/issues/4410)

**Reply by martinschneider01** [reply]
Sorry just saw your reply!

Thanks a lot, I think that should do it!
